addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5271,51 +5314,129 @@ registry.layout_column = SnippetOptionWidget.extend({
      * last x columns.
      *
      * @private
-     * @param {jQuery} $row - the row in which to update the columns
-     * @param {integer} count - positif to add, negative to remove
-     */
-    _updateColumnCount: async function ($row, count) {
-        if (!count) {
+     * @param {HTMLElement} rowEl - the row in which to update the columns
+     * @param {integer} nbColumns - the number of columns requested
+     */
+    async _updateColumnCount(rowEl, nbColumns) {
+        const isMobile = weUtils.isMobileView(this.$target[0]);
+        const prevNbColumns = this._getNbColumns(rowEl.children, isMobile);
+        const prevNbRows = !isMobile ? 1 : Math.ceil(rowEl.children.length / prevNbColumns);
+        // itemsDelta: positive to add, negative to remove.
+        // - Remove only if it was already on one row (otherwise, keep all
+        // items and reorganize the columns on the right amount of rows).
+        // - Add if there are more columns than elements.
+        const itemsDelta = prevNbRows === 1 || nbColumns > rowEl.children.length ?
+            nbColumns - rowEl.children.length : 0;
+
+        if (!itemsDelta && nbColumns === prevNbColumns) {
```

Comment:
It's not clear enough how `prevNbColumns` can be different from `rowEl.children.length` here and how `!itemsDelta` can be different from `nbColumns === prevNbColumns` (I understand because I know the task but just reading the comment, I am actually confused).